mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+        shuffle : bool, default=True
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            Whether or not to shuffle the data before splitting.
+            If shuffle=False
+            then stratify must be ``None``.
+        stratify : array-like, default=None
+            For the ``sklearn.model_selection.train_test_split`` documentation.
+            If not ``None``, data is split in a stratified fashion, using this
+            as the class labels.
+            Read more in the :ref:`User Guide <stratification>`.
+
+        Returns
+        -------
+        MapieCalibrator
+            The model itself.
+        """
+        self._check_method()
+        cv = check_cv(self.cv)
+        estimator = check_estimator_classification(X, y, cv, self.estimator)
+        calibrator = self._get_calibrator(self.calibrator)
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        assert type_of_target(y) in ["multiclass", "binary"]
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        self.n_features_in_ = check_n_features_in(X, cv, estimator)
+        random_state = check_random_state(random_state)
+
+        if cv == "prefit":
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.n_classes_ = len(self.classes_)
+            self.calibrators = self._fit_calibrators(
+                X, y, sample_weight, calibrator
+            )
+        if cv == "split":
+            results = get_calib_set(
+                X,
+                y,
+                sample_weight=sample_weight,
+                calib_size=calib_size,
+                random_state=random_state,
+                shuffle=shuffle,
+                stratify=stratify,
+            )
+            X_train, y_train, X_calib, y_calib, sw_train, sw_calib = results
+            X_train, y_train = indexable(X_train, y_train)
+            y_train = _check_y(y_train)
+            sw_train, X_train, y_train = check_null_weight(
+                sw_train,
+                X_train,
+                y_train
+            )
+            self.n_classes_ = len(np.unique(y_train))
+            self.estimator = fit_estimator(
+                clone(estimator), X_train, y_train, sw_train,
+            )
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.calibrators = self._fit_calibrators(
+                X_calib, y_calib, sw_calib, calibrator
+            )
+        return self
+
+    def predict_proba(
+        self,
+        X: ArrayLike,
+    ) -> NDArray:
+        """
+        Prediction of the calibrated scores using fitted classifer and
+        calibrator.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes)
+            The calibrated score for each max score and zeros at every
```

Comment:
The calibrated score at the maximum score

I wonder if this method will suit to other callibration method than top-level.

Reply:
Why zero instead of nan? And do you need an array whereas you only store a single value per line?

Reply:
The goal was to keep the prediction output such that one could also get the `np.argmax` from this. For me, this would be the best way to fit any method. If you have any other ideas, I'm all ears.